File: auction-web/src/types/auction.ts
export interface Auction {
  id: number
  seller_id: number
  title: string
  description: string | null
  category_id: number | null
  starting_price: number
  current_price: number
  bid_increment: number
  bid_count: number
  status: 'draft' | 'active' | 'ended' | 'sold' | 'cancelled'
  start_time: string
  end_time: string
  auto_extend: boolean
  winner_id: number | null
  created_at: string
  images: AuctionImage[]
}

export interface AuctionImage {
  id: number
  url: string
  sort_order: number
  is_primary: boolean
}

export interface AuctionListResponse {
  items: Auction[]
  total: number
  page: number
  page_size: number
}

export interface AuctionCreate {
  title: string
  description?: string
  category_id?: number
  starting_price: number
  reserve_price?: number
  bid_increment: number
  start_time: string
  end_time: string
  auto_extend?: boolean
}

File: auction-api/src/schemas/auction.py
from pydantic import BaseModel
from datetime import datetime
from decimal import Decimal


class AuctionCreate(BaseModel):
    title: str
    description: str | None = None
    category_id: int | None = None
    starting_price: Decimal
    reserve_price: Decimal | None = None
    bid_increment: Decimal
    start_time: datetime
    end_time: datetime
    auto_extend: bool = True


class AuctionUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    category_id: int | None = None
    reserve_price: Decimal | None = None


class AuctionResponse(BaseModel):
    id: int
    seller_id: int
    title: str
    description: str | None
    category_id: int | None
    starting_price: Decimal
    current_price: Decimal
    bid_increment: Decimal
    bid_count: int
    status: str
    start_time: datetime
    end_time: datetime
    auto_extend: bool
    winner_id: int | None
    created_at: datetime
    images: list[dict] = []

    class Config:
        from_attributes = True


class AuctionListResponse(BaseModel):
    items: list[AuctionResponse]
    total: int
    page: int
    page_size: int
